modular_skyrat/modules/modular_implants/code/soulcatcher/attachable_soulcatcher.dm

```diff
@@ -0,0 +1,57 @@
+/datum/component/soulcatcher/attachable_soulcatcher
+	max_souls = 1
+	communicate_as_parent = TRUE
+	removable = TRUE
+
+/datum/component/soulcatcher/attachable_soulcatcher/New()
+	. = ..()
+	var/obj/item/parent_item = parent
+	if(!istype(parent_item))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
+	RegisterSignal(parent, COMSIG_CLICK_CTRL_SHIFT, PROC_REF(bring_up_ui))
+
+/// Adds text to the examine text of the parent item, explaining that the item can be used to enable the use of NIFSoft HUDs
+/datum/component/soulcatcher/attachable_soulcatcher/proc/on_examine(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER
+	examine_text += span_cyan("[source] has a soulcatcher attached to it, <b>Ctrl+Shift+Click</b> to use it.")
+
+/datum/component/soulcatcher/attachable_soulcatcher/proc/bring_up_ui(datum/source, mob/user)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(ui_interact), user)
+
+/datum/component/soulcatcher/attachable_soulcatcher/Destroy(force)
+	UnregisterSignal(parent, COMSIG_ATOM_EXAMINE)
+	UnregisterSignal(parent, COMSIG_CLICK_CTRL_SHIFT)
+	return ..()
+
+/datum/component/soulcatcher/attachable_soulcatcher/remove_self()
+	var/obj/item/parent_item = parent
+	var/turf/drop_turf = get_turf(parent_item)
+	new /obj/item/attachable_soulcatcher (drop_turf)
+
+	return ..()
+
+/obj/item/attachable_soulcatcher
+	name = "mini-soulcatcher kit"
```

Comment:
name = "Poltergeist-Type RSD
desc = "This device, a polymorphic nanomachine net, wraps around objects of most sizes and allows them to function as a container for Resonance. The soul in question within the vessel is imbued much like it would be in a body or a normal Soulcatcher, perceiving the world and even speaking out of their new form. The nanomachine net of the device allows for the consciousness to somewhat manipulate their container, but any large-scale movement is out of the question."